### DISCUSSION TITLE
Replacing AwsClientProviderImpl using @Provider annotation

### DISCUSSION THREAD

**Discussion by almeicar** [eclipse-edc/Connector#4609] [Q&A] [answered]
Hey!
We have created a custom extension in order to override the AwsClientProviderImpl using the @Provider(isDefault = true) annotation. However this approach is not working and it seems our version is not being picked up.
Is there something we're missing or should this be done in a different way?

`@Provider(isDefault = true)
 public AwsClientProvider awsClientProvider(ServiceExtensionContext context) {
  ...
  }`

**Comment by paullatzelsperger** [comment]
please _carefully_ read the [documentation about dependency injection](https://eclipse-edc.github.io/documentation/for-contributors/control-plane/dependency-injection/). What you are trying to do is only possible if the `S3CoreExtension` is removed from the classpath because you cannot replace a `@Provider` method.

`isDefault` means "use this if no other is available"

**Reply by almeicar** [reply]
Thank you for the quick reply!
Do you know of any other approach we can follow as we still would like to use the s3 extension functionality, just with a different provider?

**Reply by paullatzelsperger** [reply]
the `AwsClientProviderImpl` is not meant to be replaced. Why do you want to do that? I.e. what does your client do that the stock one doesn't?

**Reply by almeicar** [reply]
We basically want the fix that exists for 0.10.1 (https://github.com/eclipse-edc/Technology-Aws/commit/a4a95b069c8002a01f5b4f5e8653f595fe2b7310) but we're using a lower version and can't update. Our goal is to disable the caching in our custom provider.

**Reply by paullatzelsperger** [reply, marked as answer]
ok, then simply adding the `@Provider(isDefault = true)` won't solve anything in your case. What you're effectively trying to do is backporting the fix. 

Copy-pasting the `aws-s3-core` module in your code base won't work either, because there are transitive dependencies onto it. 

Why can't you use the latest version?

**Reply by almeicar** [reply]
We are using tractusx and it uses specific edc versions and we don't have permission to upgrade unfortunately. Thank you for your help in any case :)

**Reply by paullatzelsperger** [reply]
You appear to be using a custom distribution of Tx-EDC anyway (otherwise you wouldn't be able to add any extensions), so it should be relatively simple to upgrade, at least technically. But I understand that there are some non-technical rules at play here ^^

Do you mind closing the discussion and marking the answer?
